### PR TITLE
chore(local debug): prepare task runner in vsc

### PR DESCRIPTION
**Pull request by swatDong**
This is for `npm install` prerequisites check - to run `npm install` as vscode tasks.
Added `runTask` utility to execute vsc tasks, and watch for exit code.

### CODE REVIEW

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts [hunk 1]
```diff
@@ -135,6 +179,20 @@ async function checkCustomizedPort(component: string, componentRoot: string, che
...
+function onDidStartTaskHandler(event: vscode.TaskStartEvent): void {
+  const taskId = event.execution.task?.definition?.teamsfxTaskId;
+  if (taskId !== undefined) {
+    taskStartEventEmitter.fire(taskId as string);
+  }
+}
+
+function onDidEndTaskHandler(event: vscode.TaskEndEvent): void {
+  const taskId = event.execution.task?.definition?.teamsfxTaskId;
+  if (taskId !== undefined && trackedTasks.has(taskId as string)) {
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/teamsfxTaskHandler.ts:191]: Seems missing `trackedTasks.add()` beforewards.

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts [hunk 2]
```diff
@@ -168,6 +226,12 @@ async function onDidEndTaskProcessHandler(event: vscode.TaskProcessEndEvent): Pr
   const task = event.execution.task;
   const activeTerminal = vscode.window.activeTerminal;
 
+  const taskId = task?.definition?.teamsfxTaskId;
+  if (taskId !== undefined) {
+    trackedTasks.delete(taskId as string);
+    taskEndEventEmitter.fire({ id: taskId as string, exitCode: event.exitCode });
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/teamsfxTaskHandler.ts:232]: Curious what is the behavior when a task end listener is fired twice, i.e., one on onDidEndTaskProcessHandler and the other on onDidEndTaskHandler?

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts (hunk 1 above)
Comment (swatDong) [reply packages/vscode-extension/src/debug/teamsfxTaskHandler.ts:194]: Good catch. Added `trackedTasks.add` and `trackedTasks.delete` whenever end event is fired.

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts (hunk 2 above)
Comment (swatDong) [reply packages/vscode-extension/src/debug/teamsfxTaskHandler.ts:236]: Per vscode source code, `onDidEndTaskProcessHandler` will be called first. For normal case `trackedTasks` is updated so `onDidEndTaskHandler` will not fire event.

For error case that process has not been started, only `onDidEndTaskHandler` will fire event.

For corner case that end event is fired twice, updated `runTask` to dispose listener after receiving the first event. So the second event has no effect on `runTask`.

**Pull request merged**